carma_base/Dockerfile

```diff
@@ -0,0 +1,49 @@
+#  Copyright (C) 2018 LEIDOS.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+# 
+#  http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+# CARMA Base Image Docker Configuration Script
+
+FROM ros:kinetic-ros-base 
+
+RUN apt-get update && \
+        apt-get install -y \
+        git \
+        ssh \
+        ros-kinetic-rosjava \
+        ros-kinetic-rosbridge-server \
+        sudo \
+        tmux \
+        vim \
+        nano \
+        less
+
+# Add carma user
+ENV USERNAME carma
+RUN useradd -m $USERNAME && \
+        echo "$USERNAME:$USERNAME" | chpasswd && \
+        usermod --shell /bin/bash $USERNAME && \
+        usermod -aG sudo $USERNAME && \
+        mkdir -p /etc/sudoers.d && \
+        echo "$USERNAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/$USERNAME && \
+        chmod 0440 /etc/sudoers.d/$USERNAME && \
+        usermod  --uid 1000 $USERNAME && \
+        groupmod --gid 1000 $USERNAME
```

Comment:
I guess it's safe to assume these user & group IDs haven't already been used because this is a brand-new OS with no other users at this point, right?

Reply:
Correct, the only user by default inside a container is root. The UID and GUID stuff doesn't actually matter, it's normally part of a kind of hack to fix some X-server permissions if you're sharing the X-server socket between container and host to see visual programs on the host even though they're inside your container, but we're not doing that.